### PR TITLE
[IMG-177] Android image support policy

### PR DESCRIPTION
**Pull request by JalexChen**
# Description
A brief description of the changes.
Adds Android Image Support Policy to docs

# Reasons
[IMG-177](https://circleci.atlassian.net/jira/software/c/projects/IMG/boards/683?selectedIssue=IMG-177)

- [x] Break up walls of text by adding paragraph breaks.
- [x] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [x] Keep the title between 20 and 70 characters.
- [x] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [x] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [N/A] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [x] Include relevant backlinks to other CircleCI docs/pages.

[IMG-177]: https://circleci.atlassian.net/browse/IMG-177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+contentTags:
+  platform:
+  - Cloud
+  - Server v4.x
+  - Server v3.x
```
Comment (brentmmarks) [inline jekyll/_cci2/android-images-support-policy.adoc:6]: Dont think this should effect server?

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+== Overview
+
+This document outlines the Circleci Android image release, update, and deprecation policy. This policy applies to all CircleCI Android images built for the Machine executor.
+
+[#release-policy]
+== Release policy
+
+Android images are released once a quarter, with patch releases potentially being made for security issues. These images are generally built on top of the latest version of our most recent stable base Ubuntu image with added packages for Android.
+
+We install the most up-to-date versions of each tool/package in newly built images.
```
Comment (brentmmarks) [inline jekyll/_cci2/android-images-support-policy.adoc:25]: Is there a way to make these bullet points like in doc?

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+- `default`: This image is the current stable version and of image will receive updates approximately every 3 months.
+
+- `edge`: These tags are reserved for previews of new releases, which will initially point to this tag. The `edge` tags may include incremental updates to the `current` image release, which may change without notice, and is not recommended to be used for production CI workloads. currentwill be updated with these changes after period of stability (generally an average of a week).
+
+- Date based tagging: Android images are generally released once every 3 months. The date-based tagging conventions take the format of <year>.<month>.<patch>, such as 2023.10.1. We recommend using the default version and not pinning to a date version.
+
+[#critical-cve-patches]
+== Critical CVE patches
+
+When critical CVEs are disclosed that affect the versions of the operating system or software stack in our Docker Convenience Images, we will investigate the impact that this has on our images being used within the CircleCI execution environment. If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image
```
Comment (brentmmarks) [inline jekyll/_cci2/android-images-support-policy.adoc:49]: Missing period

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+[#bug-reports-issues-and-prs]
+== Bug reports, issues, and PRs
+
+You can file a [ticket]https://support.circleci.com/hc/en-us/requests/new with CircleCI Support for any issues or bugs found with the remote Docker images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.
```
Comment (brentmmarks) [inline jekyll/_cci2/android-images-support-policy.adoc:54]: Please file a [Support Ticket with CircleCI Support](https://support.circleci.com/hc/en-us/requests/new) for any issues or bugs found with the Android VM (Machine) Images.

- Current mentioned remote docker

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+| Android
+| API Level 27 (Android 8); Retain Q4 release of previous two years
+
+|===
+
+Example: When Android 15 releases:
+
+[.table.table-striped]
+[cols=2*, options="header", stripes=even]
```
Comment (brentmmarks) [inline jekyll/_cci2/android-images-support-policy.adoc:77]: This does not match what is in policy we got sign off for from support and others?

**Review by ryanbourdais — COMMENTED** [review]
Looks good to me, aside from the notes @brentmmarks made

File: jekyll/_cci2/android-images-support-policy.adoc (hunk 4 above)
Comment (JalexChen) [reply jekyll/_cci2/android-images-support-policy.adoc:54]: will do

File: jekyll/_cci2/android-images-support-policy.adoc (hunk 3 above)
Comment (JalexChen) [reply jekyll/_cci2/android-images-support-policy.adoc:49]: will do

File: jekyll/_cci2/android-images-support-policy.adoc (hunk 2 above)
Comment (JalexChen) [reply jekyll/_cci2/android-images-support-policy.adoc:25]: will do

File: jekyll/_cci2/android-images-support-policy.adoc (hunk 5 above)
Comment (JalexChen) [reply jekyll/_cci2/android-images-support-policy.adoc:77]: will do

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+This document outlines the Circleci Android image release, update, and deprecation policy. This policy applies to all CircleCI Android images built for the Machine executor.
+
+[#release-policy]
+== Release policy
+
+Android images are released once a quarter, with patch releases potentially being made for security issues. These images are generally built on top of the latest version of our most recent stable base Ubuntu image with added packages for Android.
+
+- We install the most up-to-date versions of each tool/package in newly built images.
+- We aim to package 6 levels of the Android API within each image release.
+- As this image is not specifically designed for the Android Gradle Plugin (AGP), some updates may come later than desired. However, we aim to follow compatibility matrices for https://docs.gradle.org/current/userguide/compatibility.html and https://developer.android.com/build/releases/gradle-plugin#updating-gradle.
```
Comment (brentmmarks) [inline jekyll/_cci2/android-images-support-policy.adoc:25]: Can you fix formatting here for Gradle and AGP

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+[#critical-cve-patches]
+== Critical CVE patches
+
+When critical CVEs are disclosed that affect the versions of the operating system or software stack in our Docker Convenience Images, we will investigate the impact that this has on our images being used within the CircleCI execution environment. If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.
+
+[#bug-reports-issues-and-prs]
+== Bug reports, issues, and PRs
+
+Please file a [Support Ticket with CircleCI Support](https://support.circleci.com/hc/en-us/requests/new) for any issues or bugs found with the Android VM (Machine) Images. Our Support Team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.
```
[the remark this answers is not in the document]
Comment (JalexChen) [reply jekyll/_cci2/android-images-support-policy.adoc:50]: @brentmmarks 
1) technically, the link formatting is standard and correct per docs, but i can change it to `link:URL[label]`
2) are you sure we are linking to discuss for this? our policy doc in confluence is pointing to the support link

[the remark this answers is not in the document]
Comment (brentmmarks) [reply jekyll/_cci2/android-images-support-policy.adoc:50]: was meant as example

[the remark this answers is not in the document]
Comment (brentmmarks) [reply jekyll/_cci2/android-images-support-policy.adoc:50]: https://github.com/circleci/circleci-docs/commit/2f5c5377d30b27bb6fa9f81c67a83c0eb822746f

**Review by brentmmarks — APPROVED** [review]

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+We will keep a total of 7 images at maximum, with a deprecation cycle happening once per year. The table below shows what images will be kept in each cycle. Generally we will aim to start EOL process within 3 months of new version release.
+
+Current Deprecation (Jan 2024):
+
+[.table.table-striped]
+[cols=2*, options="header", stripes=even]
+|===
+| Release Cycle
+| Image Status
```
Comment (rosieyohannan) [inline jekyll/_cci2/android-images-support-policy.adoc:63]:
```suggestion
| Release cycle
| Image status
```

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+| Current Year
+| Keep all quarterly images of this year; Keep Q4 (October) image release of last 2 years
```
Comment (rosieyohannan) [inline jekyll/_cci2/android-images-support-policy.adoc:66]:
```suggestion
| Current year
| Keep all quarterly images of this year. Keep Q4 (October) image release of the last 2 years
```

File: jekyll/_cci2/android-images-support-policy.adoc [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+|===
+
+When an image is selected for deprecation and removal, we will create an announcement on our Discuss forum, along with reaching out via email to developers who have requested one of the deprecated images in their recent jobs.
+
+We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release
```
Comment (rosieyohannan) [inline jekyll/_cci2/android-images-support-policy.adoc:72]:
```suggestion
We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release.
```

**Review by rosieyohannan — APPROVED** [review]

**Pull request merged**